week-6/Homework/mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,75 @@
-// Write your code here
+const mainContainer = document.createElement("div");
+document.body.appendChild(mainContainer);
+
+const imgContainer = document.createElement("img");
+
+const slideImages = ["img-01.jpg", "img-02.jpg", "img-03.jpg", "img-04.jpg"];
+
+imgContainer.src = slideImages[0];
+
+let i = 0;
+let j;
+function moveForward() {
+  imgContainer.setAttribute("src", `${slideImages[i]}`);
+  i++;
+  j = i - 2;
+  if (i === slideImages.length) {
+    i = 0;
+  } else {
+    i = i;
+  }
+}
+
+function moveBack() {
+  imgContainer.setAttribute("src", `${slideImages[j]}`);
+  j--;
+  i = j + 2;
+  j = j < 0 ? slideImages.length - 1 : j;
+}
+
+mainContainer.appendChild(imgContainer);
+
+// auto back button
+const autoBackBtn = document.createElement("button");
+autoBackBtn.innerText = "Auto-Back";
+document.body.appendChild(autoBackBtn);
+
+// back button
+
+const backBtn = document.createElement("button");
+backBtn.innerText = "Back";
+document.body.appendChild(backBtn);
+backBtn.addEventListener("click", moveBack);
+
+autoBackBtn.addEventListener("click", function () {
+  var back = setInterval(moveBack, 2000);
+  stopBtn.addEventListener("click", function () {
+    clearInterval(back);
+  });
```

Comment:
before setting the intervals you need to clear them, as clicking autoforward, autobackward, set you in a loop between two images

Reply:
I think I cleared them now, but still some bugs need to sort. Like auto buttons try to run each time clicked before resetting the previous click.